plugins/package-managers/node/src/main/kotlin/Yarn.kt

```diff
@@ -88,12 +93,27 @@ class Yarn(
 
         val process = run(workingDir, "info", "--json", packageName)
 
-        val data = Json.parseToJsonElement(process.stdout).jsonObject["data"]?.also {
+        val data = parseYarnInfo(process.stdout)?.jsonObject?.get("data")?.also {
             yarnInfoCache.write(packageName, it.toString())
-        } ?: checkNotNull(Json.parseToJsonElement(process.stderr).jsonObject["data"]).also {
+        } ?: checkNotNull(parseYarnInfo(process.stderr)?.jsonObject?.get("data")).also {
             logger.warn { "Error running '${process.commandLine}' in directory $workingDir: $it" }
         }
 
         return parsePackageJson(data)
     }
 }
+
+/**
+ * Parse the given [output] of a Yarn _info_ command to a [JsonElement] that can be further processed. The output is
+ * typically a JSON object with the metadata of the package that was queried. However, under certain circumstances,
+ * Yarn may return multiple JSON objects separated by newlines; for instance, if the operation is retried due to
+ * network problems. This function filters for the object with the data based on the _type_ field. Result is *null* if
+ * no matching object is found or the input is not valid JSON.
+ */
+internal fun parseYarnInfo(output: String): JsonElement? =
+    runCatching {
+        ByteArrayInputStream(output.toByteArray()).use { inputStream ->
+            Json.decodeToSequence<JsonObject>(inputStream)
+                .firstOrNull { elem -> elem["type"]?.jsonPrimitive?.content == "inspect" }
```

Comment:
I think it's fine to just use `it` here for shortness.

Reply:
As `.jsonPrimitive` throws if "type" is not a `JsonPrimitve`, the "loop" will be exited in this case without checking the remaining elements. I agree it's a very low risk that a "type" is not a `JsonPrimitve`, but for a tad safer code we could use

    .firstOrNull { (it["type"] as? JsonPrimitive)?.content == "inspect" }